networkx/algorithms/tests/test_core.py

```diff
@@ -67,6 +71,17 @@ def test_core_number2(self):
         assert nodes_equal(nodes_by_core[1], [1, 3])
         assert nodes_equal(nodes_by_core[2], [2, 4, 5, 6])
 
+    def test_core_number_self_loop(self):
+        self_loop = nx.cycle_graph(3)
+        self_loop.add_edge(0, 0)
+        with pytest.raises(nx.NetworkXError):
+            nx.core_number(self_loop)
```

Comment:
A very minor nit, but using `self_loop` as a variable name for the graph object could be a bit confusing (at face value `self_loop` seems like a tighter fit as the name of an edge rather than a graph containing self loops).

IMO it'd be a minor improvement to the readability to stick with the standard naming convention:

```suggestion
    def test_core_number_self_loop(self):
        G = nx.cycle_graph(3)
        G.add_edge(0, 0)
        with pytest.raises(nx.NetworkXError, match="Input graph has self loops"):
            nx.core_number(G)
```

Note also the suggestion to add a check for the exception message.

Reply:
The same advice applies to the test of `onion_layers` as well.